src/plugin/Component.h

```diff
@@ -118,6 +118,25 @@ class Component {
      */
     zeek::Tag Tag() const;
 
+    /**
+     * Returns true if the component is currently enabled and hence
+     * available for use.
+     */
+    bool Enabled() const { return enabled; }
+
+    /**
+     * Enables or disables this component. Derived classes may override this if
+     * they need to initiate additional actions, but must then call the base
+     * class version.
```

Comment:
Well, in principle yes, but (a) unfortunately pretty much nothing of the Zeek-side analyzer life-cycle logic is documented anywhere anyways and (b) while this moves `SetEnabled()` into the base class, it doesn't change semantics re/ `Initialize` compared to what we already had before for analyzers. Given those, I'm not really inclined to go further here, as I don't think it would help much.